### PR TITLE
Refactor redux store setup for tests

### PR DESCRIPTION
**Pull request by mconiglio**
#### :page_facing_up: Description:

Currently, the http client interceptors are getting the auth headers from the redux store. The problem is that, for the tests, we are using a different store, which is built accepting an initial state. So this is causing that the authentication headers are never being sent on the tests requests.

This PR tackles this issue by implementing a way on which the interceptors are created dynamically and setup using the store. Also, it unifies the store logic to be shared between the app and the tests configuration.

---

#### :heavy_check_mark: Tasks:

* Refactor the http client to have the interceptors added dynamically with a specific store
* Delete the old store setup for tests and use the same one as the app
* Create the `setupTests.js` file to setup global stuff like a callback to remove the interceptors added to the http client after each test
* Fix a couple of endpoint mocks after fixing the root problem

---

@loopstudio/react-devs

### CODE REVIEW

File: src/services/middlewares/auth.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    (error) => {
+      if (!error.response) {
+        return Promise.reject({ errors: ['Connection error'] });
+      }
+
+      if (error.response.status === 401) {
+        store.dispatch(clearSession());
+      }
+
+      return Promise.reject(error.response.data);
```
Comment (mwolman) [inline src/services/middlewares/auth.js:30]:
```suggestion
    ({ response }) => {
      if (!response) {
        return Promise.reject({ errors: ['Connection error'] });
      }

      if (response.status === 401) {
        store.dispatch(clearSession());
      }

      return Promise.reject(response.data);
```

File: src/setupTests.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import httpAdapter from 'axios/lib/adapters/http';
+import httpClient from 'services/httpClient';
```
Comment (mwolman) [inline src/setupTests.js:2]:
```suggestion

import httpClient from 'services/httpClient';
```

File: src/services/middlewares/auth.js (hunk 1 above)
Comment (ValeFranchi) [reply src/services/middlewares/auth.js:30]: I remember this was done on purpose, as above is showing `response` also, maybe you prefer to change it anyway but i prefer to keep it the way it is to make the differece more clear.

File: src/testUtils/mocks/auth.js [hunk 3]
```diff
@@ -52,17 +52,13 @@ export const mockUpdateUserSuccess = (user, passwordCheck) => {
...
   const body = passwordCheck ? { user, passwordCheck } : { user };
 
-  return baseMock
-    .patch('/user', decamelizeKeys(body))
-    .query({ locale: 'en' })
-    .reply(400, {
-      errors: ['Some scary error'],
-    });
+  return baseMock.patch('/user', decamelizeKeys(body)).reply(400, {
+    errors: ['Some scary error'],
```
Comment (lucaleivaloop) [inline src/testUtils/mocks/auth.js:62]: 👻

Comment (mconiglio) [reply src/testUtils/mocks/auth.js:62]: 👻

File: src/setupTests.js (hunk 2 above)
Comment (mconiglio) [reply src/setupTests.js:2]: why?

Comment (mwolman) [reply src/setupTests.js:2]: We are always leaving an empty line between the imports of files that are ours and externals, right? 🤔

Comment (mconiglio) [reply src/setupTests.js:2]: Ok. Didn't notice that. I thought it only was for absolute/relative imports

**Pull request merged**